ee/clickhouse/queries/funnels/funnel_trends_persons.py

```diff
@@ -1,6 +1,48 @@
-from ee.clickhouse.queries.funnels.base import ClickhouseFunnelBase
+from rest_framework.exceptions import ValidationError
 
+from ee.clickhouse.queries.funnels.funnel_trends import TIMESTAMP_FORMAT, ClickhouseFunnelTrends
+from ee.clickhouse.queries.util import get_trunc_func_ch
+from posthog.constants import DROP_OFF, ENTRANCE_PERIOD_START, OFFSET
+from posthog.models.person import Person
 
-class ClickhouseFunnelTrendsPersons(ClickhouseFunnelBase):
-    def get_query(self):
-        pass
+
+class ClickhouseFunnelTrendsPersons(ClickhouseFunnelTrends):
+    def get_query(self) -> str:
+        drop_off = self._filter.drop_off
+        if drop_off is None:
+            raise ValidationError(f"Filter parameter {DROP_OFF} must be provided and a bool for funnel trends persons!")
+
+        entrance_period_start = self._filter.entrance_period_start
+        if not entrance_period_start:
+            raise ValidationError(
+                f"Filter parameter {ENTRANCE_PERIOD_START} must be provided and a datetime for funnel trends persons!"
+            )
+
+        step_counts_query = self.get_step_counts_without_aggregation_query(
+            specific_entrance_period_start=entrance_period_start
+        )
+        # Expects multiple rows for same person, first event time, steps taken.
+        self.params.update(self.funnel_order.params)
+
+        _, reached_to_step_count_condition, did_not_reach_to_step_count_condition = self.get_steps_reached_conditions()
+        self.params[OFFSET] = self._filter.offset
+
+        query = f"""
```

Comment:
Now, this can instead be:
```
return FUNNEL_PERSONS_BY_STEP_SQL.format(
            offset=self._filter.offset,
            steps_per_person_query=steps_per_person_query,
            persons_steps=did_not_reach_to_step_count_condition if drop_off else reached_to_step_count_condition,
        )
```

(optional/separate PR: we could make offset a query param as well, good shout)

Reply:
Good point, refactored 👌